### DISCUSSION TITLE
Ruffle games on Android (It kinda worked?)

### DISCUSSION THREAD

**Discussion by wearyexcellance** [ruffle-rs/ruffle#6497] [Ideas]
Hello there, I just tried to port a flash game using Ruffle and Android Studio to Mobile. 
**How It made it work?**
I actually downloaded ruffle html example, changed the swf link then I took the swf game page to an open-source web browser (It is called Sweb). So I modified the homepage to the ruffle game.  Thankfully it worked
**Disadvantages :** 
-Pretty slow for some games
-No game buttons
-Works only with internet (not fully offline)

![image](https://user-images.githubusercontent.com/101978004/159169960-2e1e91cb-da33-408e-84f5-319d19feec91.png)

**Comment by foumart** [comment]
You can make it work offline if you set up a service worker (if Sweb supports it). Otherwise I did not understood exactly how you made it work, did you pack the open source browser inside the APK?

Here is what I've tested so far:
I tried to run a Ruffle game in a progressive web application (PWA) and it worked really nice. Upon visiting the web page where the game is embedded with Android Chrome, it prompts you to install the PWA, after which the game works like an app with offline support.
Another way I tried is to use Cordova and load the webpage+ruffle in the web view - it also worked pretty good.
